pennylane/data/dataset.py

```diff
@@ -174,6 +182,8 @@ def _read_file(filepath):
     def read(self, filepath, lazy=False):
         """Loads data from a saved file to the current dataset.
 
+        Note that this method assumes that the file contents are of the form ``{attribute_name: attribute_value,}``.
```

Comment:
Should we add another note for `hamiltonian` dictionaries?

Reply:
I don't think so. `*_hamiltonian.dat` files are _not_ of the form `{attribute_name: attribute_value}`, so the note already addresses this!

Reply:
I meant that reading them would require additional conversion (and they technically are of that type lol).

Reply:
I think it's an implementation detail that users don't need to know about! Plus - I should clarify I'm using "attribute" as a reserved word here to mean a value in a dataset. Anyway, it's all been fixed. Let me update this comment.